EvidenceApi/V1/UseCase/UpdateDocumentSubmissionStateUseCase.cs

```diff
@@ -66,6 +66,13 @@ public async Task<DocumentSubmissionResponse> ExecuteAsync(Guid id, DocumentSubm
                 throw new BadRequestException("This state is invalid");
             }
 
+            if ((documentSubmission.State == SubmissionState.Approved ||
+                documentSubmission.State == SubmissionState.Rejected) &&
+                (request.State == "APPROVED" || request.State == "REJECTED"))
+            {
+                throw new BadRequestException("Document has already been approved/rejected");
```

Comment:
You could change this exception message so it includes the current state, instead of saying it may be one of those. For example `Document has already been ${documentSubmission.State}`

Reply:
Good point! Addressed